modemdconfig.go

```diff
@@ -0,0 +1,56 @@
+/*
+modemd - Communicates with USB modems
+Copyright (C) 2019, The Cacophony Project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package main
+
+import (
+	"io/ioutil"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type ModemdConfig struct {
+	ModemsConfig      []ModemConfig `yaml:"modems"`
+	TestHosts         []string      `yaml:"test-hosts"`
+	TestInterval      int           `yaml:"test-interval"`
+	PowerPin          string        `yaml:"power-pin"`
+	InitialOnTime     int           `yaml:"initial-on-time"`
```

Comment:
You make these be based on time.Duration with string inputs so the duration in the configuration file can be `30s` (seconds) or `2m` (minutes) (whatever time.ParseDuration expects. This makes the config easier to read and use and avoids confusion about units. Have these intervals already in time.Duration makes them easier to use in the code too.

Example:
- Helper type: https://github.com/jumptrading/influx-spout/blob/master/config/duration.go
- Use in config struct: https://github.com/jumptrading/influx-spout/blob/master/config/config.go#L52
- Use of the value in the config struct: https://github.com/jumptrading/influx-spout/blob/master/listener/listener.go#L239